src/main/tray.ts

```diff
@@ -159,6 +177,38 @@ export class Tray {
       this.settings = cfg;
       this.settingsChanged();
     });
+
+    Electron.ipcMain.on('update-network-status', (_, status: boolean) => {
+      this.handleUpdateNetworkStatus(status).catch((err:any) => {
+        console.log('Error updating network status: ', err);
+      });
+    });
+    setInterval(() => {
+      this.checkNetworkStatus().catch((err:any) => {
+        console.log('Error updating network status: ', err);
+      });
+    }, 15_000);
```

Comment:
I'm not sure polling is a reasonable solution here — this means the machine will never go to sleep (hence we why also only check for images when on the images page).

We will need a better solution here (though I'm not sure it exists); alternatively, we shouldn't have a constantly-checking thing (and instead of a manually-triggered one for startup, and for manually switching to the diagnostics page).

Reply:
OK, I'm going to pull the polling thing out, and rely on the network-on/off event to trigger processing (when the window is up).

Reply:
Next solution is to refresh the connection status when the main window is redisplayed. 

But as long as the menu is closed, the tray menu item won't update when network connectivity
changes.

I'm ok with this -- it's mostly an edge case. Someone with a very flaky connection would find lots of
things not working well, not just the rancher desktop tray menu.